subprojects/docs/src/samples/ivy-publish/java-multi-project/output-ivy.xml

```diff
@@ -10,16 +10,21 @@
     <description>The first project</description>
   </info>
   <configurations>
+    <conf name="apiElementsJavadoc" visibility="public"/>
     <conf name="compile" visibility="public"/>
     <conf name="default" visibility="public" extends="runtime"/>
     <conf name="runtime" visibility="public"/>
+    <conf name="runtimeElementsSources" visibility="public"/>
   </configurations>
   <publications>
-    <artifact name="project1" type="sources" ext="jar" conf="compile" m:classifier="sources"/>
     <artifact name="project1" type="jar" ext="jar" conf="compile,runtime"/>
+    <artifact name="project1" type="jar" ext="jar" conf="apiElementsJavadoc" m:classifier="javadoc"/>
+    <artifact name="project1" type="jar" ext="jar" conf="runtimeElementsSources" m:classifier="sources"/>
   </publications>
   <dependencies>
     <dependency org="junit" name="junit" rev="4.12" conf="runtime-&gt;default"/>
     <dependency org="org.gradle.sample" name="project2" rev="1.0" conf="runtime-&gt;default"/>
+    <dependency org="junit" name="junit" rev="4.12" conf="runtimeElementsSources-&gt;default"/>
```

Comment:
I wonder about the choice of doing secondary variants given this. The reason we have dependencies here is because of that relationship between the `sources` and `runtimeElements`.